components/header-bar/src/online-status.js

```diff
@@ -0,0 +1,31 @@
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from './online-status.styles.js'
+
+/** A badge to display online/offline status in the header bar */
+export function OnlineStatus({ status, info, mobile }) {
+    // Convert to title case
+    const displayStatus = status.replace(/(^|\s)\S/g, c => c.toUpperCase())
```

Comment:
I prefer to put `useOnlineStatus` and `status` inside `OnlineStatus` and keep it isolated there. Usage from the HeaderBar component would be: `<OnlineStatus />`.

Calling `useOnlineStatus` on each render of the HeaderBar even when `pwaEnabled` is `false` hits every app that is not a PWA.

If we see that calling `OnlineStatus` multiple times is a problem we can look into optimizations, but to start with we should place the code cohesively if we can.

Reply:
That sounds fine to me 👍 

Here's another consideration that may be interesting that came up when I was working on [adding the `lastOnline` status](https://github.com/dhis2/app-runtime/pull/974) to the `useOnlineStatus` hook though:
Since the `lastOnline` date is useful to track across sessions and apps, it might be useful for even non-PWA apps to have the `useOnlineStatus` hook somewhere so that the `lastOnline` date is also updated from there.  Ultimately, I don't think this would make an important difference in the information that users are getting from the lastOnline date (i.e. I think the most important information comes from the online status _while using PWA apps_), so I think it's fine to go with what you suggested 👍